protoc.mk

```diff
@@ -0,0 +1,28 @@
+# This Makefile makes it easy for the language bindings
+# to download and install the protoc binary.
+
+PROTOC_OS := $(shell uname -s)
+ifeq (Darwin,$(PROTOC_OS))
+PROTOC_OS := osx
+endif
+
+PROTOC_ARCH := $(shell uname -m)
+ifeq (i386,$(PROTOC_ARCH))
+PROTOC_ARCH := x86_32
+endif
+
+PROTOC := ./protoc
```

Comment:
I don't like the way we build protoc all the time. For C++ projects like Mesos/grpc, it might already bundle protoc and probably do not want to build protoc again. I'd suggest we allow downstream project to pass in a path to protoc (pre-existed at some location).

If you do need a `*.mk` file for modularization, please put that under `build/make` subdirectory. I might be interested in adding CMake support under `build/cmake`